backend/src/chats/chats.gateway.ts

```diff
@@ -43,7 +59,38 @@ export class ChatsGateway {
       chatId: data.chatId,
     });
 
-    // Send the new message back to all clients
-    this.server.emit('newMessage', newMessage);
+    this.server.to(`chat:${data.chatId}`).emit('newMessage', newMessage);
+  }
+
+  @SubscribeMessage('joinChat')
+  async handleJoinChat(
+    @ConnectedSocket() client: Socket,
+    @User() user: UserEntity,
+    @MessageBody() chatId: number,
+  ) {
+    client.join(`chat:${chatId}`);
+    client.to(`chat:${chatId}`).emit('userJoined', user);
+  }
```

Comment:
Essa segunda linha é só uma resposta pro cliente dizendo que deu certo, acho que não precisa mandar um novo evento. 
Como no nosso front a gente tá sendo otimista e presumindo que vai dar certo: 
```ts
socket.emit("joinChat", chat.id);
```
Sugiro nem mandar resposta e só dar join
```suggestion
  ) {
    client.join(`chat:${chatId}`);
  }
```
Assim a gente não confunde pensando que esse `userJoined` é um evento que outros devem escutar